packages/next/server/render.tsx

```diff
@@ -1285,6 +1316,86 @@ function mergeResults(chunks: Array<RenderResult>): RenderResult {
   }
 }
 
+function multiplexResult(result: RenderResult): RenderResult {
+  const chunks: Array<string> = []
+  const subscribers: Set<Observer<string>> = new Set()
+  let streamResult:
+    | {
+        kind: 'COMPLETE'
+      }
+    | {
+        kind: 'FAILED'
+        error: Error
+      }
+    | null = null
+
+  result({
+    next(chunk) {
+      chunks.push(chunk)
+      subscribers.forEach((subscriber) => {
+        try {
+          subscriber.next(chunk)
+        } catch {}
+      })
+    },
+    error(error) {
+      if (!streamResult) {
+        streamResult = {
+          kind: 'FAILED',
+          error,
+        }
+        subscribers.forEach((subscriber) => {
+          try {
+            subscriber.error(error)
+          } catch {}
+        })
+        subscribers.clear()
+      }
+    },
+    complete() {
+      if (!streamResult) {
+        streamResult = {
+          kind: 'COMPLETE',
+        }
+        subscribers.forEach((subscriber) => {
+          try {
+            subscriber.complete()
+          } catch {}
+        })
+        subscribers.clear()
+      }
+    },
+  })
+
+  return (subscriber) => {
+    let completed = false
+    process.nextTick(() => {
+      for (const chunk of chunks) {
+        if (completed) {
+          return
+        }
+        subscriber.next(chunk)
+      }
+
+      if (!completed) {
+        if (!streamResult) {
+          subscribers.add(subscriber)
+        } else {
+          if (streamResult.kind === 'FAILED') {
+            subscriber.error(streamResult.error)
+          } else {
+            subscriber.complete()
```

Comment:
not sure if it need a try-catch here like [line](https://github.com/vercel/next.js/pull/28085/files#diff-63632e6e580e69692941c89bab0a0e0436ade3f9543ef610925f795928d9d5fdR1361)